tasks/kernel_matrix_testing/stacks.py

```diff
@@ -191,12 +191,22 @@ def launch_stack(ctx, stack, ssh_key, x86_ami, arm_ami):
 
     provision = ""
     if remote_vms_in_config(vm_config):
-        provision = "--provision"
-
-    env_vars = ' '.join(env)
-    ctx.run(
-        f"{env_vars} {prefix} inv -e system-probe.start-microvms {provision} --instance-type-x86={X86_INSTANCE_TYPE} --instance-type-arm={ARM_INSTANCE_TYPE} --x86-ami-id={x86_ami} --arm-ami-id={arm_ami} --ssh-key-name={ssh_key} --infra-env=aws/sandbox --vmconfig={vm_config} --stack-name={stack} {local}"
-    )
+        provision = "--provision-instance"
+
+    args = {
+        "--provision-microvms" if provision_microvms else "",
+        f"--instance-type-x86={X86_INSTANCE_TYPE}",
+        f"--instance-type-arm={ARM_INSTANCE_TYPE}",
+        f"--x86-ami-id={x86_ami}",
+        f"--arm-ami-id={arm_ami}",
+        f"--ssh-key-name={ssh_key}",
+        "--infra-env=aws/sandbox",
+        f"--vmconfig={vm_config}",
+        f"--stack-name={stack}",
+        local,
+        provision,
+    }
```

Comment:
Why using a hashset instead of a list ? I think it is better to preserve the arguments order

Reply:
Ah good catch. I meant it to be a list!